### PR TITLE
GG-7517 lib updates to make ready for play3 upgrades

### PR DESCRIPTION
**Pull request by jinbal-hmrc**

### CODE REVIEW

**Review by ModisR — DISMISSED** [review]
LGTM

File: build.sbt [hunk 1]
```diff
@@ -12,28 +12,29 @@ lazy val taxYear = (project in file("."))
...
+  "com.vladsch.flexmark" % "flexmark-all"   % "0.36.8" % Test,
+  "org.pegdown"          % "pegdown"        % "1.6.0"  % Test
 )
 
 libraryDependencies := {
   CrossVersion.partialVersion(scalaVersion.value) match {
     case Some((2, scalaMajor)) if scalaMajor <= 11 =>
       libraryDependencies.value ++ Seq(
-        "org.scalatestplus"      %% "scalatestplus-scalacheck"  % "3.1.0.0-RC2" % Test
+        "org.scalatestplus" %% "scalatestplus-scalacheck" % "3.1.0.0-RC2" % Test
```
Comment (tim-holbrook) [inline build.sbt]: Not sure we should be using RC's here, can we go back to the last stable version?

Comment (ModisR) [reply build.sbt]: Ooh, that's a good point. Didn't consider it. According to the [official Scalatest docs](https://www.scalatest.org/user_guide/property_based_testing), the last stable version seems to be `scalacheck-1-17` with Scala Test v3.2.17.0. However, [this is only available from Scala 2.12](https://central.sonatype.com/search?q=scalacheck-1-17). For Scala 2.11, [it appears you have to use Scala Check 1.15](https://central.sonatype.com/artifact/org.scalatestplus/scalacheck-1-15_2.11), for which [the latest stable Scala Test is v3.2.3.0](https://central.sonatype.com/artifact/org.scalatestplus/scalacheck-1-15_2.11/versions).

Having said that, Scala 2.11 isn't included in the `crossScalaVersions` setting so it's strange that this code is here at all. Maybe we can delete this `case` altogether?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ tax-year
 
 This is a UK tax year utility. Therefore, it uses the `Europe/London` timezone.
 
+Change History
```
Comment (tim-holbrook) [inline README.md:7]: Please change to the ticket text, namely:

v4.0 - 05 Dec 23
- update Scala 2.13 version
- update library dependencies
- add 'Change History' section to README

v3.3.0 - 01 Aug 23
- add README instructions

v3.2.0 - 06 May 22
- update Scala version

v2.0.0 - 07 Apr 22
- drop redundant scalatestplus-play dependency

**Review by tim-holbrook — APPROVED** [review]

**Pull request merged**